appinventor/appengine/src/com/google/appinventor/client/youngandroid/YoungAndroidFormUpgrader.java

```diff
@@ -511,6 +511,19 @@ private static int upgradeBarcodeScannerProperties(Map<String, JSONValue> compon
       // The UseExternalScanner property was added.
       srcCompVersion = 2;
     }
+    if (srcCompVersion < 3) {
+      // The UseExternalScanner property was set to True by default.
+      if (componentProperties.containsKey("UseExternalScanner")){
+        String value = ((ClientJsonString)componentProperties.get("UseExternalScanner")).getString();
+        if (value.equals("True")){
```

Comment:
The default value was True, so this will never be the case because we don't store the default value. Instead, you want to remove the key if the value is "False".